src/Storage/Doctrine/PhpcrOdm/ContentTypeDriver.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Symfony\Cmf\Component\ContentType\Storage\Doctrine\PhpcrOdm;
+
+use Doctrine\Common\Persistence\Mapping\Driver\MappingDriver;
+use Symfony\Cmf\Component\ContentType\MappingBuilderCompound;
+use Symfony\Cmf\Component\ContentType\MappingInterface;
+use Symfony\Cmf\Component\ContentType\FieldRegistry;
+use Symfony\Cmf\Component\ContentType\Mapping\StringMapping;
+use Doctrine\Common\Persistence\Mapping\ClassMetadata;
+use Symfony\Cmf\Component\ContentType\MappingRegistry;
+use Symfony\Cmf\Component\ContentType\MappingBuilder;
+use Symfony\Cmf\Component\ContentType\Mapping\IntegerMapping;
+
+class ContentTypeDriver implements MappingDriver
+{
+    private $registry;
+    private $mappings;
+    private $initialized = false;
+    private $mappingRegistry;
+
+    public function __construct(FieldRegistry $registry, MappingRegistry $mappingRegistry)
+    {
+        $this->registry = $registry;
+        $this->mappingRegistry = $mappingRegistry;
+    }
+
+    private function init()
+    {
+        if ($this->initialized) {
+            return;
+        }
+
+        foreach ($this->registry->all() as $fieldName => $field) {
+            $mappingBuilder = new MappingBuilder($this->mappingRegistry);
+            $mapping = $field->getMapping($mappingBuilder);
```

Comment:
We should wrap the mapping builder in another class and handle the subsequent logic below so that we always get a MappingInterface instance.